lib/channel-postmessage/src/index.ts

```diff
@@ -205,11 +205,18 @@ const getEventSourceUrl = (event: MessageEvent) => {
     try {
       return element.contentWindow === event.source;
     } catch (err) {
-      const src = element.getAttribute('src');
-      const { origin } = new URL(src);
+      //
+    }
+
+    const src = element.getAttribute('src');
+    let origin;
 
-      return origin === event.origin;
+    try {
+      ({ origin } = new URL(src));
+    } catch (err) {
+      ({ origin } = new URL(src, document.location));
```

Comment:
Since the second argument will be called implicitly with `document.location.toString()` so wouldn't `new URL(document.location)` be sufficient in this case?

Reply:
Yeah I guess..

Reply:
I'm posting my proposal here but I leave it up to you folks 👍 

```js
  try {
    ({ origin } = new URL(src, document.location));
  } catch (err) {
    // throws when src contains invalid absolute URL
    console.error(`Failed to obtain iframe origin from src: ${src}`);
    throw err;
  }
```

Reply:
Does adding the document.location do nothing when the passed src is already a full url?

Reply:
@ndelangen yes, you can find a note about the behavior in [docs](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL)

> If url is a relative URL, base is required, and will be used as the base URL. If url is an absolute URL, a given base will be ignored.

Reply:
TIL, thanks!